_posts/2024-03-19-credential-store-for-passwords.adoc

```diff
@@ -0,0 +1,171 @@
+```
+Once you have added your passwords to the credential store, you can enable history again using the command below: 
+```
+[standalone@localhost:9999 /] history --enable
+```
+
+== Update the Keystore Credentials
+Now, we can edit our keystore to use the credential store instead of the clear-text password: 
+```
+/subsystem=elytron/key-store=serverKS:write-attribute(name=credential-reference, value={store=myCredStore, alias=kspass})
+```
+Now if we use the read-resource function, we can no longer see the password: 
+```
+{
+    "outcome" => "success",
+    "result" => {
+        "alias-filter" => undefined,
+        "credential-reference" => {
+            "store" => "myCredStore",
+            "alias" => "kspass"
+        },
+        "path" => "server.keystore",
+        "relative-to" => "jboss.server.config.dir",
+        "required" => false,
+        "provider-name" => undefined,
+        "providers" => undefined,
+        "type" => "PKCS12"
+    },
+    "response-headers" => {"process-state" => "reload-required"}
+}
+```
+
+== Automatic Credential Store Update
+WildFly also allows us to automatically add a new password to a previously created credential-store when specifying the password for a reference. We can automatically add a new password by using the credential-reference attribute for a resource and by specifying the clear-text password there using the following commands: 
+```
+/subsystem=elytron/key-store=serverKS:write-attribute(name=credential-reference, value={store=myCredStore, alias=example, clear-text=secret})
+```
+This will produce the an output like this: 
+```
+{
+    "outcome" => "success",
+    "result" => {"credential-store-update" => {
+        "status" => "new-entry-added",
+        "new-alias" => "example"
+    }},
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+```
+As you can see from the output, the credential-store has been updated to add a new credential and that is being used for the keystore now. If you reload the server and call the read-resource function on the keystore, this will be the output: 
+```
+{
+    "outcome" => "success",
+    "result" => {
+        "alias-filter" => undefined,
+        "credential-reference" => {
+            "store" => "myCredStore",
+            "alias" => "example"
+        },
+        "path" => "server.keystore",
+        "relative-to" => "jboss.server.config.dir",
+        "required" => false,
+        "provider-name" => undefined,
+        "providers" => undefined,
+        "type" => "PKCS12"
+    }
+}
+```
+Notice how even though we specified the clear-text password when updating the credentials, it does not show up here. Instead, we can see the name of the credential-store and the alias listed under credential-reference. 
+
+== Remove Unused Credentials
+If you are no longer using an and would like to remove it from the credential store, then you can do that using the following command: 
```

Comment:
s/If you are no longer using an and /If you are no longer using an alias and

Reply:
fixed!